src/Utilities/Math.hpp

```diff
@@ -162,3 +164,23 @@ template <typename T>
 constexpr T sgn(const T& val) {
   return sgn_detail::sgn(val, std::is_signed<T>{});
 }
+
+/// \ingroup UtilitiesGroup
+/// \brief Raises type T to the integer power n.
+template <typename T>
+T integer_pow(const T& x, const int e) {
+  ASSERT(e >= 0, "Negative powers are not implemented");
+  int ecount = e;
+  int bitcount = 1;
+  while (ecount >>= 1) {
+    ++bitcount;
+  }
+  auto result = make_with_value<T>(x, 1.);
```

Comment:
This is bad, it's doing a memory allocation... I think the better thing to do is to plug into Blaze directly similar to how `src/DataStructures/Blaze/StepFunction.hpp` works. Then provide an overload for double/float. There was also some concern of temporaries in SpEC: that only matters if you are operating on the low-level type, and I would expect the switch-case to be faster than this more complicated implementation. Basically, the way I think we should do this is add the low-level support rather than trying to figure out some way around it at the high level.

Reply:
Are my new commits roughly what you meant? Im not sure what you mean by low level in this context.